### PR TITLE
Feat: LCFS - FSE date range default #714

### PR DESCRIPTION
**Pull request by prv-proton**
- #714 (FSE date range defaulted)
- DateRangeCellEditor bug fix for UTC date format.
- fix for ```request: Request``` missing positional parameter updates on all the views files on backend, which was causing issues on some of the endpoints
- Add Floating button to scroll up or down with a click on compliance report.
- CommonArray cell Renderer styling fix in case of read only ag-grid
- compliance period endpoint to be cached 
- once the new compliance report button is clicked by choosing the compliance year this button will show loading and disable any click actions on it.
- Show Introduction accordion expanded only when a new report is created.

### CODE REVIEW

File: frontend/src/views/ComplianceReports/ComplianceReports.jsx [hunk 1]
```diff
@@ -19,11 +19,11 @@ import { useCreateComplianceReport } from '@/hooks/useComplianceReports'
 // internal components
 import { reportsColDefs } from './components/_schema'
 import { NewComplianceReportButton } from './components/NewComplianceReportButton'
-import Loading from '@/components/Loading'
 
 export const ComplianceReports = () => {
   const { t } = useTranslation(['common', 'report'])
   const [alertMessage, setAlertMessage] = useState('')
+  const [isButtonLoading, setIsButtonLoading] = useState(false)
```
Comment (hamed-valiollahi) [inline frontend/src/views/ComplianceReports/ComplianceReports.jsx:26]: It's nice to have it 👍  Thanks for adding.

File: frontend/src/views/ComplianceReports/EditViewComplianceReport.jsx [hunk 2]
```diff
@@ -22,24 +24,51 @@ import { useGetComplianceReport } from '@/hooks/useComplianceReports'
...
 
   const [alertMessage, setAlertMessage] = useState('')
   const [alertSeverity, setAlertSeverity] = useState('info')
   const alertRef = useRef()
 
   const { compliancePeriod, complianceReportId } = useParams()
-  const location = useLocation()
+  const [isAtTop, setIsAtTop] = useState(true);
+
+  const scrollToTopOrBottom = () => {
```
Comment (hamed-valiollahi) [inline frontend/src/views/ComplianceReports/EditViewComplianceReport.jsx:44]: It's also very useful for these kinds of pages with lots of content.

**Review by hamed-valiollahi — CHANGES_REQUESTED** [review]
I checked out your branch and encountered a few `roles_required` issues. After fixing those issues, I was able to test the changes.

The default date range for the FSE calendar is fine, but when I try to change it by clicking on the calendar icon, I get an error.

**Comment by prv-proton** [comment]
Thanks for the comments, have resolved all the issues.

**Pull request merged**